presto-main/src/main/java/com/facebook/presto/sql/relational/optimizer/ExpressionOptimizer.java

```diff
@@ -56,6 +56,7 @@
 import static com.google.common.base.Predicates.instanceOf;
 import static com.google.common.collect.ImmutableList.toImmutableList;
 
+@Deprecated
 public class ExpressionOptimizer
```

Comment:
- This class is not used by any prod code anymore. Consider moving it as a test class `TestingExpressionOptimizer`.
- Actually rather than making it in test scope, we should *completely* remove it. `ExpressionOptimizer` is used by some critical tests like `TestExpressionInterpreter`. It is dangerous to have this class not running in production yet checking some critical paths. Behavior will become inconsistent if we still rely on this for tests.

Reply:
Yes, removing it will be the goal, but I haven't included it in this PR.